python-threatexchange/threatexchange/signal_type/signal_base.py

```diff
@@ -143,29 +148,31 @@ class TextHasher(FileHasher):
     """
 
     @classmethod
+    @abc.abstractmethod
     def hash_from_str(cls, text: str) -> str:
         """
         Get a string representation of the hash from a string.
 
         If a hash cannot be generated, empty string should be returned.
         """
-        raise NotImplementedError
+        pass
 
     @classmethod
     def hash_from_file(cls, file: pathlib.Path) -> str:
         return cls.hash_from_str(file.read_text())
 
 
-class MatchesStr:
+class MatchesStr(abc.ABC):
     @classmethod
+    @abc.abstractmethod
     def matches_str(cls, signal: str, haystack: str) -> SignalComparisonResult:
         """
         Compare the distance of two hashes, the key operation for matching.
 
         Note that this can just be a reference/helper, and the efficient
         version of the algorithm can live in the index class.
         """
-        raise NotImplementedError
+        pass
```

Comment:
does the addition of `@abc.abstractmethod` mean this will still throw or is this a logic change as well?

Reply:
Even better! It now fails typing and also throws on class construction.